src/components/CodeSnippet/CodeSnippet.tsx

```diff
@@ -0,0 +1,143 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+import type { ReactNode } from "react";
+
+export type DropdownOptionPros = {
+  label: string;
+  value: string | number;
+  selected?: boolean;
+  disabled?: boolean;
+};
+
+export type CodeSnippetDropdownProps = {
+  onChange?: (evt: React.SyntheticEvent) => void;
+  options: Array<DropdownOptionPros>;
+};
+
+export const CodeSnippetDropdown = ({
+  options,
+  onChange,
+}: CodeSnippetDropdownProps): JSX.Element => {
+  return (
+    <select className="p-code-snippet__dropdown" onChange={onChange}>
+      {options.map(({ label, value, ...props }) => (
+        <option value={value} key={value || label} {...props}>
+          {label}
+        </option>
+      ))}
+    </select>
+  );
+};
+
+CodeSnippetDropdown.props = {
+  onChange: PropTypes.func,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string.isRequired,
+      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    })
+  ).isRequired,
+};
+
+export const CodeSnippetBlock = ({
+  code,
+  title,
+  numbered = false,
+  icon = null,
+  isWrapped = false,
+  dropdowns,
+}: CodeSnippetBlockProps): JSX.Element => {
+  let className = "p-code-snippet__block";
+  let numberedCode;
+
+  if (numbered) {
+    className += "--numbered";
+
+    // wrap code lines in spans (and preserve the whitespace)
+    const lines = code.split(/\r?\n/);
+    numberedCode = lines.map((line, i) => (
+      <React.Fragment key={`p-code-snippet__line-${i}`}>
+        <span className="p-code-snippet__line">{line}</span>
+        {"\n"}
```

Comment:
I think using display block on lines doesn't work well with `pre` formatting. Because both blocks create new lines, and new lines in `pre` are preserved.
So, what we are trying to do here is wrap each line in span, but preserve the whitespaces (new lines).